mockrxandroidble/src/test/groovy/com/polidea/rxandroidble2/mockrxandroidble/RxBleClientMockTest.groovy

```diff
@@ -356,28 +389,221 @@ public class RxBleClientMockTest extends ElectricSpecification {
                 .take(1)
                 .map { scanResult -> scanResult.getBleDevice() }
                 .flatMap { rxBleDevice -> rxBleDevice.establishConnection(false) }
-                .flatMapSingle { rxBleConnection -> rxBleConnection.readCharacteristic(characteristicUUID) }
+                .flatMapSingle { rxBleConnection -> rxBleConnection.readCharacteristic(characteristicUUIDNoCallback) }
                 .map { data -> new String(data) }
                 .test()
 
         then:
         testSubscriber.assertValue("Polidea")
     }
 
+    def "should return characteristic data via callback"() {
+        when:
+        characteristicReadSubject.take(1).subscribe { result -> result.success("PolideaCB".getBytes()) }
```

Comment:
This line should be put in `given:` clause I think — this is a test setup. We could even make a convenience function to be used like:
```groovy
given:
def testDataString = "PolideaCB"
nextCharacteristicReadWillResult { result -> result.success(testDataString.getBytes()) }
// (...)
then:
testSubscriber.assertValue(testDataString)
```
this would be equivalent of:
```
characteristicReadSubject.take(1).subscribe { result -> result.success(testDataString.getBytes()) }
```
The same goes to other test cases